conda_libmamba_solver/solver.py

```diff
@@ -682,7 +694,17 @@ def _parse_problems(cls, problems: str) -> Mapping[str, MatchSpec]:
                 conflicts.append(MatchSpec(pin_b))
             elif "is excluded by strict repo priority" in line:
                 # package python-3.7.6-h0371630_2 is excluded by strict repo priority
-                conflicts.append(cls._str_to_matchspec(words[2]))
+                conflicts.append(cls._matchspec_from_error_str(words[1]))
+            elif line == "unsupported request":
+                # libmamba v2 has this message for package not found errors
+                # we need to double check with the explained problem
```

Comment:
This would be excellent to clean up somehow

Reply:
Unfortunately we can't get the "explained problems" in a structured way directly. We could get the graph (`Unsolvable.problems_graph()`) but then we need to redo the logic to obtain the actual conflicts, and that involves several `Tree*` objects only available in the C++ layer. So the simplest thing here is to parse line by line and trim the hierarchy indicators 😬